kafka-rest/src/main/java/io/confluent/kafkarest/entities/ConsumerGroupLag.java

```diff
@@ -0,0 +1,171 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.entities;
+
+import com.google.auto.value.AutoValue;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class ConsumerGroupLag {
+
+  ConsumerGroupLag() {
+  }
+
+  public abstract Long getMaxLag();
+
+  public abstract Long getTotalLag();
+
+  public abstract String getClusterId();
+
+  public abstract String getConsumerGroupId();
+
+  public abstract String getMaxLagClientId();
+
+  public abstract String getMaxLagConsumerId();
+
+  public abstract Optional<String> getMaxLagInstanceId();
+
+  public abstract String getMaxLagTopicName();
+
+  public abstract Integer getMaxLagPartitionId();
+
+  public static Builder builder() {
+    return new AutoValue_ConsumerGroupLag.Builder();
+  }
+
+  @AutoValue
+  abstract static class Offset {
+
+    abstract String getTopicName();
+
+    abstract String getConsumerId();
+
+    abstract String getClientId();
+
+    abstract Optional<String> getInstanceId();
+
+    abstract int getPartitionId();
+
+    abstract long getCurrentOffset();
+
+    abstract long getEndOffset();
+
+    private long getLag() {
+      return getEndOffset() - getCurrentOffset();
+    }
+
+    private static Builder builder() {
+      return new AutoValue_ConsumerGroupLag_Offset.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+
+      abstract Builder setTopicName(String topicName);
+
+      abstract Builder setConsumerId(String consumerId);
+
+      abstract Builder setClientId(String clientId);
+
+      abstract Builder setInstanceId(Optional<String> instanceId);
+
+      abstract Builder setPartitionId(int partitionId);
+
+      abstract Builder setCurrentOffset(long currentOffset);
+
+      abstract Builder setEndOffset(long endOffset);
+
+      abstract Offset build();
+    }
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    private Long maxLag;
+    private long totalLag = 0;
+
+    private final Set<String> consumers = new HashSet<>();
+
+    Builder() {
+    }
+
+    public void addOffset(
+        String topicName,
+        String consumerId,
+        String clientId,
+        Optional<String> instanceId,
+        int partitionId,
+        long currentOffset,
+        long endOffset
+    ) {
+
+      Offset offset =
+          Offset.builder()
+              .setTopicName(topicName)
+              .setConsumerId(consumerId)
+              .setClientId(clientId)
+              .setInstanceId(instanceId)
+              .setPartitionId(partitionId)
+              .setCurrentOffset(currentOffset)
+              .setEndOffset(endOffset)
+              .build();
```

Comment:
What do you need Offset for? The only method you ever call from it is `getLag()`, which is literally just `endOffset - currentOffset`. You can just add here `long lag = endOffset - currentOffset` and then erase that entire class.